src/bda/plone/discount/calculator.py

```diff
@@ -120,6 +134,11 @@ def lookup_cascade(self, context):
             group_lookup = self.group_lookup_factory(
                 context, self.date, group)
             lookups.append(group_lookup)
+        if self.coupon:
+            try:
+                lookups.append(self.coupon_lookup_factory(context, self.date, self.coupon))
```

Comment:
To make code more readable, I would split this long append, as it is e.g. in line 134.